src/components/TrapFocus/tests/TrapFocus.test.tsx

```diff
@@ -169,6 +189,136 @@ describe('<TrapFocus />', () => {
+      (document.addEventListener as jest.Mock).mockRestore();
+      (document.removeEventListener as jest.Mock).mockRestore();
+    });
+
+    interface HandlerMap {
+      [eventName: string]: any;
+    }
+
+    const listenerMap: HandlerMap = {};
+
+    it('does nothing if trapping is false', () => {
+      const trapFocus = mountWithAppProvider(<TrapFocus trapping={false} />);
+      const activeElement = document.activeElement;
+
+      listenerMap.keydown({keyCode: Key.Tab, element: trapFocus});
+
+      expect(activeElement).toBe(document.activeElement);
+      expect(focusUtils.focusFirstKeyboardFocusableNode).not.toHaveBeenCalled();
+      expect(focusUtils.focusLastKeyboardFocusableNode).not.toHaveBeenCalled();
+    });
+
+    it('prevents default if the target is the last element and the shift key is not pressed', () => {
+      const preventDefaultSpy = jest.fn();
+
+      const trapFocus = mountWithAppProvider(
+        <TrapFocus>
+          <button />
+          <input />
+        </TrapFocus>,
+      );
+
+      listenerMap.keydown({
+        keyCode: Key.Tab,
+        target: lastNode(trapFocus),
+        preventDefault: preventDefaultSpy,
+      });
+
+      expect(preventDefaultSpy).toHaveBeenCalled();
+    });
+
+    it('focuses the first keyboard focusable node', () => {
+      const trapFocus = mountWithAppProvider(
+        <TrapFocus>
+          <button />
+          <input />
+        </TrapFocus>,
+      );
+
+      listenerMap.keydown({
+        keyCode: Key.Tab,
+        target: lastNode(trapFocus),
+        preventDefault: noop,
+      });
+
+      expect(document.activeElement).toBe(firstNode(trapFocus));
+      expect(focusUtils.focusFirstKeyboardFocusableNode).toHaveBeenCalled();
+    });
+
+    it('prevents default if the target is the first element and the shift key is pressed', () => {
+      const preventDefaultSpy = jest.fn();
+
+      const trapFocus = mountWithAppProvider(
+        <TrapFocus>
+          <button />
+          <input />
+        </TrapFocus>,
+      );
+
+      listenerMap.keydown({
+        keyCode: Key.Tab,
+        shiftKey: Key.Shift,
+        target: firstNode(trapFocus),
+        preventDefault: preventDefaultSpy,
+      });
+
+      expect(preventDefaultSpy).toHaveBeenCalled();
+    });
+
+    it('focuses the last keyboard focusable node', () => {
+      const trapFocus = mountWithAppProvider(
+        <TrapFocus>
+          <button />
+          <input />
+        </TrapFocus>,
+      );
+
+      listenerMap.keydown({
+        keyCode: Key.Tab,
+        shiftKey: Key.Shift,
+        target: firstNode(trapFocus),
+        preventDefault: noop,
+      });
+
+      expect(document.activeElement).toBe(lastNode(trapFocus));
+      expect(focusUtils.focusLastKeyboardFocusableNode).toHaveBeenCalled();
+    });
+  });
 });
 
 function noop() {}
+
+function firstNode(element: ReactWrapper) {
+  const elementNode = element.getDOMNode();
+
+  if (Array.isArray(elementNode))
```

Comment:
If the `element` has a fragment as the highest parent (in this case `Focus`), then getting the DOM node of this will be an array because you cannot have one element if the fragment contains multiple.